staging/src/k8s.io/client-go/README.md

```diff
@@ -0,0 +1,188 @@
+* The `plugin/pkg/client/auth` packages contain optional authentication plugins for obtaining credentials from external sources.
+* The `transport` package is used to set up auth and start a connection.
+* The `tools/cache` package is useful for writing controllers.
+
+### Versioning
+
+- For each `v1.x.y` Kubernetes release, the major version (first digit)
+would remain `0`.
+
+- Bugfixes will result in the patch version (third digit) changing. PRs that are
+cherry-picked into an older Kubernetes release branch will result in an update
+to the corresponding branch in `client-go`, with a corresponding new tag
+changing the patch version.
+
+#### Branches and tags.
+
+We will create a new branch and tag for each increment in the minor version
+number. We will create only a new tag for each increment in the patch
+version number. See [semver](http://semver.org/) for definitions of major,
+minor, and patch.
+
+The HEAD of the master branch in client-go will track the HEAD of the master
+branch in the main Kubernetes repo.
+
+#### Compatibility: your code <-> client-go
+
+The `v0.x.y` tags indicate that go APIs may change in incompatible ways in
+different versions.
+
+See [INSTALL.md](INSTALL.md) for guidelines on requiring a specific
+version of client-go.
+
+#### Compatibility: client-go <-> Kubernetes clusters
+
+Since Kubernetes is backwards compatible with clients, older `client-go`
+versions will work with many different Kubernetes cluster versions.
+
+We will backport bugfixes--but not new features--into older versions of
+`client-go`.
+
+
+#### Compatibility matrix
+
+|                               | Kubernetes 1.23 | Kubernetes 1.24 | Kubernetes 1.25 | Kubernetes 1.26 | Kubernetes 1.27 | Kubernetes 1.28 |
+| ----------------------------- | --------------- | --------------- | --------------- | --------------- | --------------- | --------------- |
+| `kubernetes-1.23.0`/`v0.23.0` | ✓               | +-              | +-              | +-              | +-              | +-              |
+| `kubernetes-1.24.0`/`v0.24.0` | +-              | ✓               | +-              | +-              | +-              | +-              |
+| `kubernetes-1.25.0`/`v0.25.0` | +-              | +-              | ✓               | +-              | +-              | +-              |
+| `kubernetes-1.26.0`/`v0.26.0` | +-              | +-              | +-              | ✓               | +-              | +-              |
+| `kubernetes-1.27.0`/`v0.27.0` | +-              | +-              | +-              | +-              | ✓               | +-              |
+| `kubernetes-1.28.0`/`v0.28.0` | +-              | +-              | +-              | +-              | +-              | ✓               |
+| `HEAD`                        | +-              | +-              | +-              | +-              | +-              | +-              |
```

Comment:
What about the rest of the versions? Shouldnt we mention that also.

Reply:
Yes, but that should be done in a separate PR since I mainly want to mirror the current README in this PR